src/provideInlineCompletionItems.ts

```diff
@@ -31,10 +32,17 @@ export default async function provideInlineCompletionItems(
 
     const completionInfo = context.selectedCompletionInfo;
     if (completionInfo) {
-      return await getLookAheadSuggestion(document, completionInfo, position);
+      const result = await getLookAheadSuggestion(
+        document,
+        completionInfo,
+        position
+      );
+      reportSuggestionShown(document, result);
+      return result;
     }
 
     const completions = await debounceCompletions(document, position, token);
+    reportSuggestionShown(document, completions);
```

Comment:
are we sure the suggestion is indeed shown at this point (same for line 41)? I think vscode has its own debounce+cancel mechanism... this function just registers the plugin's suggestions to vscode - they decided later whether or not to present them, no?

Reply:
yes, you're right, it's the closest to rendering we have right now (the shown callback will be released by the vscode team), but from what I understand from debugging their API their debounce is before this callback is executed. so, for now, it's as good as it gets